jccli/cli/group.py

```diff
@@ -14,29 +22,159 @@ def group(ctx):
+@click.option('--system', 'type', flag_value='system_group', help='Restrict to system groups only')
+@click.pass_context
+def list_groups(ctx, type):
+    """
+    List groups
+    """
+    api2 = JumpcloudApiV2(ctx.obj.get('key'))
+    logger: Logger = ctx.obj.get('logger')
+    response = api2.get_groups(type=type)
+    serialized_response = json.dumps(response, indent=2)
+    logger.info(f"{serialized_response}")
 
 
 @group.command('delete')
 @click.option('--name', "-n", required=True, type=str, help='Name of the group')
+@click.option('--user', 'type', flag_value='user_group')
+@click.option('--system', 'type', flag_value='system_group')
 @click.pass_context
-def delete_group(ctx, name):
+def delete_group(ctx, name, type):
     """
     Delete a group
     """
-    # FIXME: Ideally, this would output JSON info of the deleted group (similar to delete-user), but at the moment it's
-    #  unclear exactly how to do that, given the API wrappers that we have.
+    logger = ctx.obj.get('logger')
+    if type is None:
+        logger.error('groups must have a type (either "user" or "system")')
+        sys.exit(1)
+    api2 = JumpcloudApiV2(ctx.obj.get('key'))
+    group = api2.get_group(group_name=name, group_type=type)
+    if group is None:
+        logger.error(f"no group found of type '{type}', {name}")
+        sys.exit(1)
+    api2.delete_group(group['id'], type)
+    logger.info(f"successfully deleted group {name}")
+
+
+@group.command('add-user')
+@click.option('--name', "-n", required=True, type=str, help='name of the group')
+@click.option('--username', '-u', required=True, type=str, help='username of user to be added')
+@click.pass_context
+def add_user(ctx, name, username):
+    """
+    Add user to group (group type is assumed to be 'user_group')
+    """
+    logger = ctx.obj.get('logger')
+    if type is None:
+        logger.error('groups must have a type (either "user" or "system")')
+        sys.exit(1)
+
+    api1 = JumpcloudApiV1(ctx.obj.get('key'))
+    api2 = JumpcloudApiV2(ctx.obj.get('key'))
+    try:
+        user = api1.get_user(username)
+    except ApiException:
+        logger.info(f"no user found named {username}")
+        sys.exit(1)
+    group = api2.get_group(group_name=name, group_type=GroupType.USER_GROUP)
+    if group is None:
+        logger.error(f"no group of type '{type}' and name '{name}' found")
+        sys.exit(1)
+    try:
+        api2.bind_user_to_group(user['id'], group['id'])
+    except ApiException:
+        logger.error(f"API error (confirm that user {username} has not already been added to group {name}")
+        sys.exit(1)
+    logger.info(f"Successfully added user '{username}' to group '{name}'")
+
+
+@group.command('list-users')
+@click.option('--name', '-n', required=True, help='name of the group')
+@click.pass_context
+def list_users(ctx, name):
+    """
+    List users in a group (group type is assumed to be 'user_group')
```

Comment:
how about `List users in a Jumpcloud 'user' group`?

Reply:
fixed